core/src/main/scala/org/apache/spark/sql/crossdata/catalog/ZookeeperStreamingCatalog.scala

```diff
@@ -0,0 +1,124 @@
+
+
+  override def dropEphemeralTable(tableIdentifier: String): Unit = {
+    val isRunning = ephemeralTableStatusDAO.dao.get(tableIdentifier).map{ tableStatus =>
+      tableStatus.status == EphemeralExecutionStatus.Started || tableStatus.status == EphemeralExecutionStatus.Starting
+    } getOrElse notFound(tableIdentifier)
+
+    if(isRunning) throw new RuntimeException("The ephemeral is running. The process should be stopped first using 'Stop <tableIdentifier>'")
+    
+    ephemeralTableDAO.dao.delete(tableIdentifier)
+    ephemeralTableStatusDAO.dao.delete(tableIdentifier)
+
+    ephemeralQueriesDAO.dao.getAll().filter( _.ephemeralTableName == tableIdentifier) foreach { query =>
+      ephemeralQueriesDAO.dao.delete(query.alias)
+    }
+  }
+
+  override def dropAllEphemeralTables(): Unit = {
+    // TODO it should be improved after changing ephemeralTableDAO.dao.deleteAll
+    Try {
+      ephemeralTableDAO.dao.deleteAll
+      ephemeralTableStatusDAO.dao.deleteAll
+      ephemeralQueriesDAO.dao.deleteAll
+    }
+  }
+
+  override def getAllEphemeralTables: Seq[EphemeralTableModel] =
+    ephemeralTableDAO.dao.getAll()
+
+
+  /**
+   * Ephemeral Queries Functions
+   */
+  override def existsEphemeralQuery(queryAlias: String): Boolean =
+    ephemeralQueriesDAO.dao.exists(queryAlias)
+
+  override def createEphemeralQuery(ephemeralQuery: EphemeralQueryModel): Either[String, EphemeralQueryModel] =
+    if(!existsEphemeralQuery(ephemeralQuery.alias))
+      Right(ephemeralQueriesDAO.dao.upsert(ephemeralQuery.alias, ephemeralQuery))
+    else Left("Ephemeral query exists")
+
+  override def getEphemeralQuery(queryAlias: String): Option[EphemeralQueryModel] =
+    ephemeralQueriesDAO.dao.get(queryAlias)
+
+  override def getAllEphemeralQueries: Seq[EphemeralQueryModel] =
+    ephemeralQueriesDAO.dao.getAll()
+
+  override def dropEphemeralQuery(queryAlias: String): Unit =
+    ephemeralQueriesDAO.dao.delete(queryAlias)
+
+  override def dropAllEphemeralQueries(): Unit = ephemeralQueriesDAO.dao.deleteAll
+
+  /**
+   * Ephemeral Status Functions
+   */
+  override def createEphemeralStatus(tableIdentifier: String,
+                                     ephemeralStatusModel: EphemeralStatusModel): EphemeralStatusModel =
+    ephemeralTableStatusDAO.dao.upsert(tableIdentifier, ephemeralStatusModel)
+
+  override def getEphemeralStatus(tableIdentifier: String): Option[EphemeralStatusModel] =
+    ephemeralTableStatusDAO.dao.get(tableIdentifier)
+
+  override def getAllEphemeralStatuses: Seq[EphemeralStatusModel] =
+    ephemeralTableStatusDAO.dao.getAll()
+
+  override def updateEphemeralStatus(tableIdentifier: String, status: EphemeralStatusModel): Unit =
+    ephemeralTableStatusDAO.dao.update(tableIdentifier, status)
+
+  override protected[crossdata] def dropEphemeralStatus(tableIdentifier: String): Unit =
+    ephemeralTableStatusDAO.dao.delete(tableIdentifier)
+
+  override protected[crossdata] def dropAllEphemeralStatus(): Unit =
```

Comment:
Is there any specific reason for having these 2 methods protected and not the rest of methods of the class?